sgt-back-end/index.js

```diff
@@ -0,0 +1,154 @@
+const pg = require('pg');
+const express = require('express');
+var app = express();
+app.use(express.json());
+
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+app.get('/api/grades/:gradeId', (req, res, next) => {
+  const gradeId = Number(req.params.gradeId);
+  if (!Number.isInteger(gradeId) || gradeId <= 0) {
+    res.status(400).json({
+      error: '"gradeId" must be a positive integer'
+    });
+    return;
+  }
+  const sql = `
+    select "gradeId",
+           "name",
+           "course",
+           "score",
+           "createdAt"
+      from "grades"
+     where "gradeId" = $1
+  `;
+  const params = [gradeId];
+  db.query(sql, params)
+    .then(result => {
+      const grade = result.rows[0];
+      if (!grade) {
+        res.status(404).json({
+          error: `Cannot find grade with "gradeId" ${gradeId}`
+        });
+      } else {
+        res.json(grade);
+      }
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({
+        error: 'An unexpected error occurred.'
+      });
+    });
+});
+
+app.get('/api/grades/', (req, res) => {
+  const sql = `
+    select *
+      from "grades"
+  `;
+  db.query(sql)
+    .then(result => {
+      const grades = result.rows;
+      res.status(200).json(grades);
+    }).catch(err => {
+      console.error(err);
+      res.status(500).json({
+        error: 'An unexpected error occurred.'
+      });
+    });
+
+});
+
+app.post('/api/grades/', (req, res) => {
+  const newGradeObj = req.body;
+  if (!newGradeObj.name || !newGradeObj.course) {
+    return res.status(400).json({ error: 'missing name, course, or score' });
+  } else if ((!Number.isInteger(newGradeObj.score) || newGradeObj.score > 100 || newGradeObj.score < 0)) {
+    return res.status(400).json({ error: 'invalid score' });
+  }
+  const params = [newGradeObj.name, newGradeObj.course, newGradeObj.score];
+  const sql = `
+  insert into "grades"
+  ("name", "course", "score")
+  values ($1, $2, $3)
+       returning *`;
+  db.query(sql, params)
+    .then(result => {
+      res.status(201).json(result.rows[0]);
+    }).catch(err => {
+      console.error(err);
+      res.status(500).json({
+        error: 'An unexpected error occurred.'
+      });
+    });
+});
+
+app.put('/api/grades/:gradeId', (req, res) => {
+  const idToUpdate = Number(req.params.gradeId);
+  if (!Number.isInteger(idToUpdate) || idToUpdate <= 0) {
+    return res.status(400).json({ error: 'id must be a positive integer' });
+  } else if (!req.body.name || !req.body.course || !req.body.score) {
+    return res.status(400).json({ error: 'missing name, course, or score' });
```

Comment:
A `score` of `0` should not trigger a `400 Bad Request`.

Reply:
<img width="865" alt="Screen Shot 2022-08-26 at 12 21 26 PM" src="https://user-images.githubusercontent.com/106217355/186976746-707e548b-9159-4aae-a32d-d66dd49a17c2.png">

Reply:
<img width="870" alt="Screen Shot 2022-08-26 at 12 23 04 PM" src="https://user-images.githubusercontent.com/106217355/186976999-c85f1a45-f300-47b3-9858-7de310a6511c.png">